### PR TITLE
Refactor shortener module

### PR DESCRIPTION
**Pull request by G07cha**
- Using request library(moved to separate function)
- All requests now returning only longUrl and shortUrl which make more easier to using this module
- Update sample file
- Update tests

**Pros:**
- Simple API
- Easy to change server

**Cons:**
- Old usage depreciated
- Other response values inaccessible(able to extend)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,15 +15,18 @@
   "engines": {
     "node": ">0.6"
```
Comment (khiro) [inline package.json]: Node version should be changed to same as "request" module.
I tested "0.6.13" and "0.7.4", both were not worked, but "0.12.7" was worked.

File: lib/shortener.js [hunk 2]
```diff
@@ -6,58 +7,47 @@ function Shortener() {}
...
-    var data = '';
-    res.on('data', function(d) {
-      data += d;
-    });
-    res.on('error', function(e) {
-      console.error(e);
-    });
-    res.on('end', function(e) {
-      callback(JSON.parse(data));
+
```
Comment (khiro) [inline lib/shortener.js:19]: Line 19, 34, 46 and 50 are only space line.
Are spaces needed ?

Comment (G07cha) [inline lib/shortener.js:19]: Not needed at all, just want to keep logical indentation.

File: package.json (hunk 1 above)
Comment (G07cha) [inline package.json]: Nice catch! Fixed.

**Comment by khiro** [comment]
Thanks, this PR is good in node "0.12.7".
I have some comments.
Please check a engine version in package.json and space indents in shortener.js.

Additionally, shortener_test.js has space and tab indents from the previous version.
I will try to fix it after I merge this PR.

File: lib/shortener.js (hunk 2 above)
Comment (khiro) [inline lib/shortener.js:19]: OK, I got it.

**Comment by G07cha** [comment]
As I set in config, it works in node "0.8.0".

**Pull request merged**